File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        ListNode* temp = node;
        node = node->next;
        while(node->next){
            temp->val = node->val;
            temp = node;
            node = node->next;
        }
        temp->val = node->val;
        temp->next = NULL;
        return;
    }
};

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode curr) {
        ListNode prev = null;
        ListNode nextNode = curr.next;
        
        while(nextNode != null){
            if(prev != null)    
                prev.val = curr.val;
            
            prev = curr;
            curr = nextNode;
            nextNode = nextNode.next;
        }
        
        if(prev != null)    
                prev.val = curr.val;
        
        prev.next = null;
    }
}